### PR TITLE
Go kit removal

### LINKED ISSUE #183: Remove go-kit
Same as what was done with workload.go but for registration and node.

### PR DESCRIPTION
**Pull request by kunzimariano**
- Please expand pkg/server/node.go to see the differences.
- Should getSpiffeIDFromCSR stay in the previously mentioned file or should we go ahead and move it into go-spiffe?

Closes #183

### CODE REVIEW

File: pkg/server/registration.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+// Implement the business logic of UpdateEntry
+func (s *registrationServer) UpdateEntry(
+	ctx context.Context, request *registration.UpdateEntryRequest) (
+	response *common.RegistrationEntry, err error) {
+	return response, err
+}
+
+// Implement the business logic of ListByParentID
```
Comment (walmav) [inline pkg/server/registration.go]: we can remove this comment and probably have a TODO

File: pkg/server/server.go [hunk 2]
```diff
@@ -154,60 +152,26 @@ func (server *Server) initDependencies() {
...
-
-	server.Config.Log.Info("Starting the Node API")
-	nodeSvc, err := node.NewService(node.Config{
-		Logger:          server.Config.Log,
-		Catalog:         server.Catalog,
-		BaseSpiffeIDTTL: server.Config.BaseSpiffeIDTTL,
-	})
-	if err != nil {
-		return err
+		return nil
```
Comment (walmav) [inline pkg/server/server.go]: should we not return the error here?

File: pkg/server/registration.go (hunk 1 above)
Comment (kunzimariano) [reply pkg/server/registration.go]: Sure thing. Just TODO?

File: pkg/server/server.go (hunk 2 above)
Comment (kunzimariano) [reply pkg/server/server.go]: Yes, fixing it.

File: pkg/server/registration.go (hunk 1 above)
Comment (walmav) [reply pkg/server/registration.go]: Yes just to make it more obvious that it needs to be implemented

File: pkg/server/server.go [hunk 3]
```diff
@@ -314,27 +278,29 @@ func (server *Server) rotateSigningCert() error {
...
-		ListByParentIDEndpoint:        registration.MakeListByParentIDEndpoint(registrationSvc),
-		ListBySelectorEndpoint:        registration.MakeListBySelectorEndpoint(registrationSvc),
-		ListBySpiffeIDEndpoint:        registration.MakeListBySpiffeIDEndpoint(registrationSvc),
-		CreateFederatedBundleEndpoint: registration.MakeCreateFederatedBundleEndpoint(registrationSvc),
-		ListFederatedBundlesEndpoint:  registration.MakeListFederatedBundlesEndpoint(registrationSvc),
-		UpdateFederatedBundleEndpoint: registration.MakeUpdateFederatedBundleEndpoint(registrationSvc),
-		DeleteFederatedBundleEndpoint: registration.MakeDeleteFederatedBundleEndpoint(registrationSvc),
+func (server *Server) getGRPCServer() (*grpc.Server, error) {
+	// TODO: Fix me after server refactor
+	crtRes, err := server.dependencies.ServerCAImpl.FetchCertificate(&ca.FetchCertificateRequest{})
```
Comment (walmav) [inline pkg/server/server.go]: Not related to the PR but wonder if we can get rid of dependencies.. right now we have init_dependencies and init_plugins, maybe we could combine them both

Comment (kunzimariano) [reply pkg/server/server.go]: After your previous comment on server.go I noticed the dependencies. I'm about to push changes that remove them.

**Comment by walmav** [comment]
💯 🥇

**Pull request merged**